### PR TITLE
Merge texture's internal slots

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
GPUTexture's internal slots contain all member parameters of
GPUTextureDescriptor object. So it makes more sense to merge
these internal slots into a GPUTextureDescriptor instance.
Other GPUObjects also follow this way and use a single descriptor
in their internal slots.

***

### :boom: Error: 500 Internal Server Error :boom: ###

[PR Preview](https://github.com/tobie/pr-preview#pr-preview) failed to build. _(Last tried on May 4, 2021, 5:10 PM UTC)_.

<details>
<summary>More</summary>

PR Preview relies on a number of web services to run. There seems to be an issue with the following one:

:rotating_light: [CSS Spec Preprocessor](https://api.csswg.org/bikeshed/) - CSS Spec Preprocessor is the web service used to build Bikeshed specs.

:link: [Related URL](https://api.csswg.org/bikeshed/?url=https%3A%2F%2Fraw.githubusercontent.com%2FRichard-Yunchao%2Fgpuweb%2Fb1cc5498a8e14d4fd97c85ca286b77aef127c55d%2Fspec%2Findex.bs&force=1&md-warning=not%20ready)

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>500 Internal Server Error</title>
</head><body>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error or
misconfiguration and was unable to complete
your request.</p>
<p>Please contact the server administrator at 
 [no address given] to inform them of the time this error occurred,
 and the actions you performed just before this error.</p>
<p>More information about this error may be available
in the server error log.</p>
<hr>
<address>Apache/2.4.10 (Debian) Server at api.csswg.org Port 443</address>
</body></html>

```

_If you don't have enough information above to solve the error by yourself (or to understand to which web service the error is related to, if any), please [file an issue](https://github.com/tobie/pr-preview/issues/new?title=Error%20not%20surfaced%20properly&body=See%20gpuweb/gpuweb%231694.)._
</details>

### CODE REVIEW

**Comment by Richard-Yunchao** [comment]
@kvark and @kainino0x , PTAL. This is the follow-up PR for the discussion at https://github.com/gpuweb/gpuweb/pull/1658#discussion_r621267273. Thanks!

**Comment by Richard-Yunchao** [comment]
FYI, @bbernhar and @RafaelCintron

**Review by kainino0x — APPROVED** [review]

File: spec/index.bs [hunk 1]
```diff
@@ -7360,12 +7337,12 @@ GPUQueue includes GPUObjectBase;
             If any of the following requirements are unmet, throw an {{OperationError}} and stop.
             <div class=validusage>
                 - |copySize|.[=Extent3D/depthOrArrayLayers=] must be `1`.
-                - |destination|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/[[textureUsage]]}}
+                - |destination|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/[[descriptor]]}}.{{GPUTextureDescriptor/usage}}
```
Comment (kvark) [inline spec/index.bs:7340]: in algorithms like this, it would make more sense to just have the |descriptor| be an argument to the algorithm, instead of repeatedly extracting it from |destination|

Comment (Richard-Yunchao) [reply spec/index.bs:7340]: I read through all changes, this is the only place where the algorithm doesn't use |destination| except its texture's descriptor info. At other places, we need to use |destination| itself. However, I feel like this algorithm is not finalized yet. For example, it may need to add 
 - [$validating GPUImageCopyTexture$](|destination|, |copySize|) returns `true`,
 during validation. 

It may need to say that 
 - copy contents into |destination|. 
after validation.

In short, I think |destination| is a must-to-have argument. But the argument |descriptor| you proposed is more like a implementation detail. If you really want to shorten the text, maybe we can add a line at the beginning of the validation as follows: 
 - Let |descriptor| be |destination|.{{GPUImageCopyTexture/texture}}.{{GPUTexture/[[descriptor]].

WDYT? (I feel like it is a temp variable and it is a implementation detail. not sure whether it is needed. )

Comment (Richard-Yunchao) [reply spec/index.bs]: Done. Right now the text is simpler and cleaner than before. 

Please help to merge this PR if you have no other concerns.

**Review by kvark — APPROVED** [review]
Thanks!
